app/src/main/java/com/example/parkingsystemjava/mvp/presenter/ParkingReservationPresenter.java

```diff
@@ -1,17 +1,85 @@
+import com.example.parkingsystemjava.utils.Constants;
+import com.example.parkingsystemjava.utils.ReservationErrorCodes;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
 
-public class ParkingReservationPresenter implements ParkingReservationContract.ReservationActivityPresenter {
+public class ParkingReservationPresenter implements ParkingReservationContract.ParkingReservationPresenter {
 
-    private ParkingReservationContract.ReservationActivityView view;
+    private final ParkingReservationContract.ParkingReservationView view;
+    private final ParkingReservationContract.ParkingReservationModel model;
 
-    public ParkingReservationPresenter(ParkingReservationContract.ReservationActivityView view) {
+    public ParkingReservationPresenter(ParkingReservationContract.ParkingReservationModel model, ParkingReservationContract.ParkingReservationView view) {
+        this.model = model;
         this.view = view;
     }
 
     @Override
-    public void showDatePicker() {
-        view.showEntryDatePicker();
+    public void showDatePicker(ListenerDateTime listenerDateTime, boolean dateSelector) {
+        view.showDatePicker(listenerDateTime, dateSelector);
+    }
+
+    @Override
+    public void addReservation(Reservation reservation, int parkingLot) {
+        model.addReservation(reservation, parkingLot);
+    }
+
+    @Override
+    public void setEntryDate(Calendar date) {
+        String stringDate = getStringFromDate(date);
+        view.showEntryDateSelected(stringDate);
+    }
+
+    @Override
+    public void setExitDate(Calendar date) {
+        String stringDate = getStringFromDate(date);
+        view.showExitDateSelected(stringDate);
+    }
+
+    @Override
+    public boolean validateReservation(Reservation reservation, int parkingLot) {
+        ReservationErrorCodes reservationCode = model.validateReservation(reservation, parkingLot);
+        if (reservationCode.equals(ReservationErrorCodes.OK)) {
+            view.showConfirmationMessage(reservationCode);
+            return true;
+        } else {
+            view.showConfirmationMessage(reservationCode);
+            return false;
+        }
+    }
+
+    @Override
+    public void validateAndSaveReservation(String entryDate, String exitDate, String keyCode, String parkingLot) {
+        Reservation reservation = new Reservation();
+        reservation.setEntryDate(entryDate);
+        reservation.setExitDate(exitDate);
+        reservation.setKeyCode(keyCode);
+        int parkingLotInt;
+        if (!parkingLot.isEmpty()) {
+            parkingLotInt = Integer.parseInt(parkingLot);
+        } else {
+            parkingLotInt = Constants.MINUS_ONE;
+        }
+        if (validateReservation(reservation, parkingLotInt)) {
+            addReservation(reservation, parkingLotInt);
+            view.closeScreen();
+        }
+    }
+
+    private String getStringFromDate(Calendar calendar) {
```

Comment:
this method can be in a DateUtils class

Reply:
done, created class and put the method in it